src/MCSSolver.py

```diff
@@ -212,13 +229,40 @@ def runDynamic(self, container, inputs):
       else:
         topEventValue[index] = self.mcsSolverMargin(inputForSolver)
 
+        sensValues = self.marginSensitivities(topEventValue, inputForSolver)
+        for key in sensitivities:
+          sensitivities[key][index] = sensValues[key]
+
     if container.tdFromPS:
       for key in container.invMapping.keys():
         container.__dict__[key] = self.timeDepData[key][0].values
 
     container.__dict__[container.timeID]     = self.timeDepData[container.timeID].values
     container.__dict__[container.topEventID] = topEventValue
 
+    if self.solver['type'] == 'margin':
+      for key in sensitivities:
+        keyID = "sens_" + str(container.invMapping[key])
+        container.__dict__[keyID] = sensitivities[key]
+
+
+  def marginSensitivities(self, MsysBase, inputDict):
```

Comment:
For now, this method is ok, I think in the future, we may need to reconsider the method to compute sensitivities.